### ISSUE TITLE
Error: "Cannot parse the file list."

### ISSUE THREAD

**Issue by sofauxboho** [pyscsd/steamCloudSaveDownloader#3]
Running the latest Docker container on the latest Unraid. Things installed smoothly and authentication worked well. Now, when I run the downloader I get this persistent error:

> / # scsd
> 2024-01-02 18:45:07 [INFO] scsd-0.0.63 started
> 2024-01-02 18:45:07 [INFO] Files will be saved to './data'
> 2024-01-02 18:45:16 [INFO] (1/169) Processing Steam Client
> 2024-01-02 18:45:19 [ERROR] Cannot parse the file list. It seems like Steam has update the webpage. Please update to the latest version or notify the author.
> 2024-01-02 18:45:19 [INFO] Retrying in 20 seconds
> 2024-01-02 18:45:39 [ERROR] Cannot parse the file list. It seems like Steam has update the webpage. Please update to the latest version or notify the author.
> 2024-01-02 18:45:39 [INFO] Retrying in 22 seconds
> 2024-01-02 18:46:02 [ERROR] Cannot parse the file list. It seems like Steam has update the webpage. Please update to the latest version or notify the author.
> 2024-01-02 18:46:02 [INFO] Retrying in 25 seconds
> 2024-01-02 18:46:27 [ERROR] Maximum attempt reached. Aborting
> 2024-01-02 18:46:27 [ERROR] Cannot parse the file list. It seems like Steam has update the webpage. Please update to the latest version or notify the author.
> / # 

So I'm letting you know as requested. Happy to provide any other info or logs that might come in handy.

Thanks for this, by the way! Steam Cloud lost my save files for a game recently and then _pushed the blank saves_ to all my computers. Steam Support was no help, and my progress is permanently lost. Excited to get this going so I can take responsibility for backing up my own data.

**Comment by sofauxboho** [comment]
And here is my (huge) games list, in case that helps. Print to PDF of https://store.steampowered.com/account/remotestorage
[steamcloudlist.pdf](https://github.com/pyscsd/steamCloudSaveDownloader/files/13813081/steamcloudlist.pdf)

**Comment by sofauxboho** [comment]
Just tried it on a second computer, a fairly typical gaming desktop running up-to-date Windows 11. Same error:

> PS C:\Users\me\working\scsd> .\scsd_v0.0.63_windows_amd64.exe -a sofauxboho
> 2024-01-02 15:21:17 [INFO] scsd-v0.0.63 started
> 2024-01-02 15:21:17 [INFO] Files will be saved to './data'
> This program will NOT save your credential locally
> Username: sofauxboho
> Password:
> 2FA code on Steam Authenticator (Please get the 5 digits code manually)
> 2FA code (case insensitive): r8qyy
> Login success. Please rerun scsd to start downloading
> PS C:\Users\me\working\scsd> .\scsd_v0.0.63_windows_amd64.exe
> 2024-01-02 15:22:03 [INFO] scsd-v0.0.63 started
> 2024-01-02 15:22:03 [INFO] Files will be saved to './data'
> 2024-01-02 15:22:13 [INFO] (1/169) Processing Steam Client
> 2024-01-02 15:22:17 [ERROR] Cannot parse the file list. It seems like Steam has update the webpage. Please update to the latest version or notify the author.
> 2024-01-02 15:22:17 [INFO] Retrying in 22 seconds
> 2024-01-02 15:22:40 [ERROR] Cannot parse the file list. It seems like Steam has update the webpage. Please update to the latest version or notify the author.
> 2024-01-02 15:22:40 [INFO] Retrying in 20 seconds
> 2024-01-02 15:23:00 [ERROR] Cannot parse the file list. It seems like Steam has update the webpage. Please update to the latest version or notify the author.
> 2024-01-02 15:23:00 [INFO] Retrying in 19 seconds
> 2024-01-02 15:23:19 [ERROR] Maximum attempt reached. Aborting
> 2024-01-02 15:23:19 [ERROR] Cannot parse the file list. It seems like Steam has update the webpage. Please update to the latest version or notify the author.
> PS C:\Users\me\working\scsd>

**Comment by hhhhhojeihsu** [comment]
Hi sofauxboho,

Thank you for your kind words.

I can't reproduce this issue on my side. I suspect that the Steam Cloud Save pages is different from expected. Could you do these two things for me:
1. Attach the contents of [saves page for Steam Client](https://store.steampowered.com/account/remotestorageapp/?appid=7) for me. More specifically, what are the strings for `Date Written`.
2. Ignore downloading saves for Steam Client(AppID 7):
Set the `mode` to `exclude`, and `list` to `7`. Please refer to [this page](https://github.com/pyscsd/steamCloudSaveDownloader/wiki/Config-File#target) for details. Please let me know if the rest of the games can be downloaded.

**Comment by sofauxboho** [comment]
Happy to. Here's a screenshot of the saves page for Steam Client:
![image](https://github.com/pyscsd/steamCloudSaveDownloader/assets/418438/5d674a3d-0411-40f9-ad2a-6b24ad3ad823)
Would you like copies of the two .vdf files available on that page as well?

I was able to exclude "7". The number of entries to process decreased from 169 to 168, and it got slightly further before hitting a similar looking error:

> PS C:\Users\me\working\scsd> .\scsd_v0.0.63_windows_amd64.exe -f .\scsd.conf
> 2024-01-03 13:19:11 [INFO] scsd-v0.0.63 started
> 2024-01-03 13:19:11 [INFO] Files will be saved to './data2'
> 2024-01-03 13:19:21 [INFO] (1/168) Processing Half-Life
> 2024-01-03 13:19:25 [ERROR] Cannot parse the file list. It seems like Steam has update the webpage. Please update to the latest version or notify the author.
> 2024-01-03 13:19:25 [INFO] Retrying in 27 seconds
> 2024-01-03 13:19:53 [ERROR] Cannot parse the file list. It seems like Steam has update the webpage. Please update to the latest version or notify the author.
> 2024-01-03 13:19:53 [INFO] Retrying in 23 seconds
> 2024-01-03 13:20:16 [ERROR] Cannot parse the file list. It seems like Steam has update the webpage. Please update to the latest version or notify the author.
> 2024-01-03 13:20:16 [INFO] Retrying in 25 seconds
> 2024-01-03 13:20:41 [ERROR] Maximum attempt reached. Aborting
> 2024-01-03 13:20:41 [ERROR] Cannot parse the file list. It seems like Steam has update the webpage. Please update to the latest version or notify the author.
> PS C:\Users\me\working\scsd>

A few notes on the above:
1) I ran this from my Windows PC rather than my Unraid server for ease of troubleshooting and to provide what may be a more typical environment.
2) I set my data directory to ./data2 in the .conf file. This was just to make it obvious if the .conf file was being read, since the data directory is shown in the initial output.
3) It did start to try and process my first game (Half-Life) this time. No files were downloaded though.

Please let me know if there's anything else I can do to help. Would a saved .html of my Steam Cloud Save page be useful?

**Comment by hhhhhojeihsu** [comment]
Hmmm that’s strange. It seems like the program fails to parse every page regardless of the games.

Yes, please provide the html file for [Steam Client save page](https://store.steampowered.com/account/remotestorageapp/?appid=7) if possible. Also please temporary disable web extensions such as ublock when doing so. There’s no need for the content of the files, just html file should suffice.

One additional question. It seems like your language is set to English, which should work. Could you check what is your primary language on [Steam](https://store.steampowered.com/account/languagepreferences)?

If all else fails I’m gonna add more debug mechanism to the build and dump the required info.

**Comment by sofauxboho** [comment]
Here's the html file for the link requested: 
[SteamCloud-appid7.html.txt](https://github.com/pyscsd/steamCloudSaveDownloader/files/13826551/SteamCloud-appid7.html.txt)
Github doesn't allow attaching html files, so please remove .txt from the end of the filename. This file was saved from Microsoft Edge since I never use it and know it doesn't have any extensions. 

Interestingly, appid 7 seems to be the Steam Client itself: https://steamdb.info/app/7/info/
The "Additional Info" mentions the Steam Deck. I did have a LCD Steam Deck and recently upgraded to an OLED Steam Deck LE. Perhaps my account has access to some funny version of the Steam Client for the new OLED or LE OLED Steam Deck, which is why I'm having troubles and others are not?

Also, confirming that my primary language on Steam is English, with no secondary languages enabled:
![image](https://github.com/pyscsd/steamCloudSaveDownloader/assets/418438/b22e163a-fa10-4cc2-9f73-b24364c515b5)

Thanks for looking into this! Happy to continue to do whatever I can.

**Comment by hhhhhojeihsu** [comment]
Well I figure it out. It’s the different time format. The program is expecting `DD MM`. But yours are showing `MM DD` instead. I have no idea Steam has different time format on different account. This shall be fixed in the next version within a few days.

One additional favor, could you check what’s the `DATE WRITTEN` field for one of the games you haven’t opened since this year(I.e. the games opened before 2024). And checked if the time format is `MM DD, YYYY` or otherwise. Thank you.

**Comment by hhhhhojeihsu** [comment]
I assume the format for game saves not modified this year would be `MM DD, YYYY`. Please check if the [newest build v0.0.65](https://github.com/pyscsd/steamCloudSaveDownloader/releases/tag/v0.0.65) works for you.

**Comment by sofauxboho** [comment]
Just checked a game that hasn't had it's save update since 2020 and it is indeed MM DD, YYYY:
![image](https://github.com/pyscsd/steamCloudSaveDownloader/assets/418438/073bb76c-0fc3-4e31-a09e-a6573143707e)

I'll go try v0.0.65 now.

**Comment by sofauxboho** [comment]
scsd v0.0.65 appears to be working smoothly on Windows 11 now!

It did throw the same error initially. I deleted all the data and conf files, reran with -a to authenticate again, and now it seems to be working normally. If anyone else has this problem you may want to try deleting your scsd.* files and trying again with the new version.

**Comment by sofauxboho** [comment]
Just updated the Docker container on my Unraid server and it went similarly. Still got the error initially. Then I deleted scsd.log, scsd.sqlite3, and session.sb and reauthed with -a. It is now working smoothly!

I suspect I only needed to delete scsd.sqlite3 but I was a bit lazy and didn't test file by file.

Now on it's on the 4th appid and still downloading smoothly. I'm going to mark this bug closed. I'll continue to monitor and if you don't hear from me again assume I'm happily using your tool to take responsibility for my own data. Thanks so much @hhhhhojeihsu!
